src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java

```diff
@@ -274,6 +274,9 @@ else if (leftOvers > 0) {
 
                 // Only set the range if the values are not equal or if the low
                 // and high are the bounds must also be the same
+                if (low > high) {
```

Comment:
Instead of this, shall we investigate why the `low` might set greater than `high`? Setting invalid `low`/`high` values might result into wrong splits.

Also, do we have a case that caused a failure at this part of the code?

Reply:
Yes, I am working on a better fix. I have a use case which was not part of the tests previously.

Reply:
Makes sense. In that case, before we merge, let's put a TODO comment (`/* TODO: */`) (so we don't forget it), and also a `log.WARN` message that indicates that `low` is greater than `high`. This will give us evidence about the possible cause in case we get odd results.

Reply:
I just return the current ranges now, same as in line 242. It solved the problem for me.